src/cli/moz-cache.js

```diff
@@ -0,0 +1,217 @@
+        'https://github.com/mozilla-b2g/B2G-manifest',
+        'https://github.com/mozilla-b2g/moztt',
+        'https://github.com/mozilla/gecko-dev',
+        'https://git.mozilla.org/build/tooltool.git',
+        'https://git.mozilla.org/external/google/gerrit/git-repo.git',
+        'https://hg.mozilla.org/projects/alder',
+        'https://hg.mozilla.org/projects/alder/',
+        'https://hg.mozilla.org/releases/mozilla-b2g34_v2_1',
+        'https://hg.mozilla.org/releases/mozilla-b2g34_v2_1/',
+        'https://hg.mozilla.org/releases/mozilla-b2g37_v2_2r',
+        'https://hg.mozilla.org/releases/mozilla-b2g37_v2_2r/',
+        'https://hg.mozilla.org/releases/mozilla-b2g37_v2_2r/',
+        'https://hg.mozilla.org/releases/mozilla-b2g44_v2_5',
+        'https://hg.mozilla.org/releases/mozilla-b2g44_v2_5/'
+    ];
+
+
+    var emulators = [
+        ['emulator_url', 'emulator-ics'],
+        ['emulator_url', 'emulator'],
+        ['emulator_url', 'emulator-jb'],
+        ['emulator_url', 'emulator-kk'],
+        ['emulator_url', 'aries'],
+        ['emulator_url', 'flame'],
+        ['emulator_url', 'flame-kk'],
+        ['emulator_url', 'nexus-4'],
+        ['emulator_url', 'emulator-l'],
+        ['emulator_url', 'dolphin'],
+        ['emulator_url', 'dolphin-512'],
+        ['g_emulator_url', 'emulator.xml'],
+        ['g_emulator_url', 'emulator-jb.xml'],
+        ['g_emulator_url', 'emulator-kk.xml'],
+        ['g_emulator_url', 'emulator-l.xml']
+    ];
+
+    var scheduler = new Scheduler({
+        // for use with taskclusterProxy
+        baseUrl: 'taskcluster/scheduler/v1'
+    });
+    var tasks = [];
+
+    for (var url of clones) {
+        tasks.push(generateCloneTaskDefinition(url));
+    }
+
+    for (var emulator of emulators) {
+        tasks.push(generateRepoCacheTaskDefinition(emulator[1], emulator[0]));
+    }
+
+
+    let {graphId, graph} = generateCacheGraph(tasks);
+    console.log(`Creating cache graph '${graphId}'`);
+    console.log(JSON.stringify(graph));
+
+    let result = await scheduler.createTaskGraph(graphId, graph);
+    console.log(JSON.stringify(result));
+}
+
+
+// To configure, define in your environment:
+//   OWNER_EMAIL
+//   TASKCLUSTER_ACCESS_TOKEN
+//   TASKCLUSTER_CLIENT_ID
+//
+// Notes:
+// * Only supports a workerType with permacreds at this time.
+
+function generateCacheGraph(tasks) {
+  let graphId = slugid.nice();
+
+  let client_id = '';
+  if (process.env.TASK_ID) {  // we're running in docker-worker in TaskCluster prod
+      client_id = 'hook-id:taskcluster/tc-vcs-refresh';
+  } else {
+      client_id = 'client-id:' + process.env.TASKCLUSTER_CLIENT_ID;
+  } // TODO maybe should just totally bail at this point
+
+  let graph = {
+      metadata: {
+        name:         'Clone Cache Graph',
+        description:  'Creates cached copies of repositories to use with taskcluster-vcs',
+        owner:        process.env.OWNER_EMAIL || "selena@mozilla.com",
```

Comment:
LOL. I know. this whole thing really aught to be reworked to use a config file. I wasn't sure what the value was though given that we're now pretty much fully into the cron solution (awaitng a dar replacement?)